nixos/modules/services/misc/owncast.nix

```diff
@@ -0,0 +1,96 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let cfg = config.services.owncast;
+in {
+
+  options.services.owncast = {
+
+    enable = mkEnableOption "owncast";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/owncast";
+      description = "The directory where owncast stores its data files.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for owncast.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "User account under which owncast runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "Group under which owncast runs.";
+    };
+
+    listen = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "127.0.0.1";
+      description = "The IP address to bind owncast to.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 80;
+      description = ''
+        TCP port where owncast web-gui listens.
+      '';
+    };
+
+    rtmp-port = mkOption {
+      type = types.port;
+      default = 1935;
+      description = ''
+        TCP port where owncast rtmp service listens.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.tmpfiles.rules = [
+      "L+ '${cfg.dataDir}/static' - - - - ${pkgs.owncast.src}/static"
+      "C '${cfg.dataDir}/webroot' 0700 - - - ${pkgs.owncast.src}/webroot"
+    ];
```

Comment:
You should probably drop this in favour of `StateDirectory` + `preStart`.

Reply:
@aanderse what about something like [this](https://github.com/ngi-nix/owncast/blob/006be90321245c904232ab61a8e71df3cefeb3b2/flake.nix#L47)? I've been working on this same package + service before knowing about this PR, and I've done things a little differently.

The benefit of my approach is that the package will create the appropriate files regardless where it's run from, and even without being a service.

Reply:
@fufexan I like that. Awesome.